### ISSUE TITLE
Feature Request: Limit user info

### ISSUE THREAD

**Issue by runningman84** [travisghansen/external-auth-server#47]
We use this config for several ingress configurations:
```
# traefik ingress
ingress.kubernetes.io/auth-type: forward
ingress.kubernetes.io/auth-url: "https://eas.example.com/verify?config_token=CONFIG_TOKEN_HERE"
ingress.kubernetes.io/auth-response-headers: X-Userinfo, X-Id-Token, X-Access-Token, Authorization
```

Recently we got this kind of error messages from nginx and apache servers:
```
400 Bad Request
Request Header Or Cookie Too Large
nginx/1.17.7
```

It turns out that users which belong to several github organisations have a lot of data in the user info part. From my point of view it would be great to get a limited Info block which would could the loginname, userid, email address, mfa, 

I checked the headers and the userinfo part alone is about 4k bytes for my user...

**Comment by travisghansen** [comment]
This is already possible (this is the 1 area though that will see breaking changes when I get around to doing a 1.0...I'm going to generalize the config a bit and make all the data be part of a single struct instead of operating on separate datapoints...doing so will allow crazy cross-check logic like compare userinfo data with token data etc).

https://github.com/travisghansen/external-auth-server/blob/master/HEADERS.md

Have a look and let me know if that's what you need. You can literally parse out whatever data you need and send it back...especially if you enable/use the raw `js` `query_engine` which let's you directly manipulate/access the data blocks and run whatever raw javascript you want.

**Comment by travisghansen** [comment]
Alternatively, if your backing services don't need the header data just remove them from the allowed headers.

**Comment by runningman84** [comment]
That sounds cool I will look into it... Right now my applications do not rely on this header so I just remove it from the list.

**Comment by Subreptivus** [comment]
> Alternatively, if your backing services don't need the header data just remove them from the allowed headers in your ingress configuration.

It's not always easy to do. For example in Istio/Envoy it's so called `ext_authz` plugin, and you don't have much control of headers in "inner" cycle (envoy <-> ext_authz service), yet you do have control over "outer" cycle (clien t<-> envoy <-> ext_authz service <-> envoy <-> destination). So you have to strip unneeded headers before it hits destination, which you can easily forget to do. And you can't decline `cookie` header (which is almost always the source of the problems with header size limits) for entering your ingress, because `eas` is actually using it.

So I would like to see this feature added as well.

BTW, `eas` itself suffers from the same workflow. If you hit `eas` with big header (for example `cookie` > 4Kb) it will return you HTTP error 400. Because `node` has default limit for header size of 4Kb. But it could be easily fixed with Environment Variable = `NODE_OPTIONS=--max-http-header-size=61440`

**Comment by travisghansen** [comment]
Ah, thanks for sharing the solution to increasing the header size! I'll document that one for sure.

@Subreptivus regarding the feature, are you referring to being able to turn on/off `eas` from returning those header back to the proxy?

**Comment by Subreptivus** [comment]
Yes, exactly.
One can narrow all the needed info using https://github.com/travisghansen/external-auth-server/blob/master/HEADERS.md to custom header (which OP is needed, and we're currently using), but you can't avoid receiving  X-Id-Token, X-Access-Token, Authorization headers.

**Comment by travisghansen** [comment]
Yeah, technically you can actually wipe them out using the custom header logic but not ideal. I plan on making all this stuff way more generic for the 1.0 release to clean it up but I'll see what can be done in the interim to make it optional.

Thanks!

**Comment by travisghansen** [comment]
This has been implemented in `v0.10.0`. Use the `features.filtered_service_headers` as documented to remove any/all of the following: `X-Userinfo, X-Id-Token, X-Access-Token`. Note that none of that alters the ability to run assertions on the values.

Any of the pieces or whole tokens can be added back using the details from the previously mentioned https://github.com/travisghansen/external-auth-server/blob/master/HEADERS.md

**Comment by travisghansen** [comment]
https://github.com/travisghansen/external-auth-server/commit/dcf8f24fccaa7cf17cfdd7dc6d005b7e54636c1b
